MatrixSDK/Room/Polls/PollAggregator.swift

```diff
@@ -0,0 +1,115 @@
+// 
+// Copyright 2021 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import Combine
+
+enum PollAggregatorError: Error {
+    case invalidPollStartEvent
+}
+
+protocol PollAggregatorDelegate: AnyObject {
+    func pollAggregatorDidStartLoading(_ aggregator: PollAggregator)
+    func pollAggregatorDidEndLoading(_ aggregator: PollAggregator)
+    func pollAggregator(_ aggregator: PollAggregator, didFailWithError: Error)
+    func pollAggregatorDidUpdateData(_ aggregator: PollAggregator)
+}
+
+class PollAggregator {
+    
+    private let session: MXSession
+    private let room: MXRoom
+    private let pollStartEvent: MXEvent
+    private let pollStartEventContent: MXEventContentPollStart
+    private let pollBuilder: PollBuilder
+    
+    private var eventListener: Any!
```

Comment:
What I really wanted was to have them all as `let non-optional` properties but also be able to construct it without having to pass everything in the initializer, so I settled on these for unwraps that would crash on runtime if any are unset.
In hindsight that wasn't the best choice, I'll figure out another way to construct them properly in the builder and have a clear interface.

Reply:
Decided to introduce an immutable protocol that just exposes everything as readonly while the builder uses the real class with all properties variable.

Reply:
Yes sure no problem we can improve that in another PR if needed. But as we don't throw errors when the parsing fails, it's probably still safer to avoid unwrap optionals if we are not sure that the property will not be set before used as it will make the app crashing.